### PR TITLE
feat: payment adapt to cosmos-sdk v0.47

### PR DESCRIPTION
**Pull request by owen-reorg**
### Description
This pr will migrate payment module to cosmos sdk v0.47.

### Rationale
To sync with cosmos sdk version.

### Example

NA

### Changes

NA

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -460,11 +460,12 @@ func New(
 		appCodec,
 		keys[paymentmoduletypes.StoreKey],
 		keys[paymentmoduletypes.MemStoreKey],
-		app.GetSubspace(paymentmoduletypes.ModuleName),
 
 		app.BankKeeper,
 		app.AccountKeeper,
 		app.SpKeeper,
+
+		authtypes.NewModuleAddress(paymentmoduletypes.ModuleName).String(),
```
Comment (yutianwu) [inline app/app.go:468]: we should use `govtypes.ModuleName` here

File: app/app.go [hunk 2]
```diff
@@ -460,11 +460,12 @@ func New(
 		appCodec,
 		keys[paymentmoduletypes.StoreKey],
 		keys[paymentmoduletypes.MemStoreKey],
```
Comment (yutianwu) [inline app/app.go:462]: memstore key is  not used

Comment (owen-reorg) [reply app/app.go]: removed

File: app/app.go (hunk 1 above)
Comment (owen-reorg) [reply app/app.go]: fixed

File: e2e/core/basesuite.go [hunk 3]
```diff
@@ -87,6 +87,18 @@ func (s *BaseSuite) SendTxBlock(msg sdk.Msg, from keys.KeyManager) *sdk.TxRespon
 	return getTxRes.TxResponse
 }
 
+func (s *BaseSuite) SimulateTx(msg sdk.Msg, from keys.KeyManager) (txRes *tx.SimulateResponse) {
+	mode := tx.BroadcastMode_BROADCAST_MODE_BLOCK
```
Comment (yutianwu) [inline e2e/core/basesuite.go:91]: it seems we don't support `block` mode now

Comment (owen-reorg) [reply e2e/core/basesuite.go]: replaced with `sync` mode

File: x/storage/keeper/payment_test.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	bankKeeper := s.depKeepers.BankKeeper
+	balances := bankKeeper.GetAllBalances(ctx, s.depKeepers.AccountKeeper.GetModuleAddress(authtypes.Minter))
+	s.T().Logf("Minter module balances: %s", balances)
+	err := bankKeeper.SendCoinsFromModuleToAccount(ctx, authtypes.Minter, s.UserAddr, coins)
+	s.Require().NoError(err)
+	balance := bankKeeper.GetBalance(ctx, s.UserAddr, "BNB")
+	s.T().Logf("s.UserAddr: %s, balance: %s", s.UserAddr, balance)
+}
+
+func (s *IntegrationTestSuiteWithoutMock) TestCreateCreateBucket_Payment() {
```
[the remark this answers is not in the document]
Comment (owen-reorg) [reply x/storage/keeper/payment_test.go:79]: payment module handles general streaming pay logic, this is storage-specific payment calculation, it's better to be here IMO

**Comment by owen-reorg** [comment]
> It's better to put cherry-pick code into another PR.

The two PRs cherry-pick into this PR are both related to the payment module, and the work of fixing test and ci are shared, so I put them together.

**Review by yutianwu — APPROVED** [review]

**Pull request merged**